04-cs-fundamentals/classroom/Binary.md

```diff
@@ -1,22 +1,277 @@
-## Required work before this class
-Before this first CS Fundamentals class, please read, view and complete the exercises on the following links:
-  + [Essential Mathematics for Software Engineers](https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/04-cs-fundamentals/classroom/Essential%20Mathematics.md)
-  + [Computer Data Basics](https://docs.google.com/document/d/1g-Dxn1rVTChBs9ZZSOgmohjP_RoWNyPJTE-m_n35aZ0/edit)
-  + [Video: Understand how memory addressing works](https://www.youtube.com/watch?v=F0Ri2TpRBBg) by Prof. Abelardo Pardo, University of Sydney
+
+## Prerequisites
+
+Before this first CS Fundamentals lesson, please read, view and complete the exercises on the following items:
+
+- [Essential Mathematics for Software Engineers](https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/04-cs-fundamentals/classroom/Essential%20Mathematics.md)
+- [Computer Data Basics](https://docs.google.com/document/d/1g-Dxn1rVTChBs9ZZSOgmohjP_RoWNyPJTE-m_n35aZ0/edit)
+- [Video: Understand how memory addressing works](https://www.youtube.com/watch?v=F0Ri2TpRBBg) by Prof. Abelardo Pardo, University of Sydney
 If you have any questions, please email or slack your instructor.
 
 ## Learning Goals
-+ High level understanding of binary
-+ Algorithm to convert from binary to decimal and vice versa
-+ High level overview of memory organization on computers
-  + Bits
-  + Bytes
-  + Computer organization: CPU-memory-buses
-  + Memory address
 
+By the end of this lesson you should be able to:
+
+- Explain the core parts of a computer and what they do
+- Describe the structure of a computer
+- Explain why computers use binary numbers
+- Compare binary numbers to decimal numbers
+- Convert between binary and decimal numbers
+- Explain how high-level languages like Ruby add a level of abstraction to make programming easier.
+
+## Typical Computer Structure
+
+A _Computer_ is a device for storing and processing data according to a series of commands or instructions provided to it.  It is normally structured with a Central Processing Unit, system memory and an I/O bus, which is often called a _motherboard_ connecting them to a variety of peripheral devices (mice, keyboards, etc).  
+
+![Computer System Example](images/computer-system.png)
+
+| Term 	| Definition 	|
+|---	|---	|
+| CPU 	| The **C**entral **P**rocessing **U**nit of a computer.  This is the main processor which executes program instructions, performs arithmetic and coordinates the other components of a computer. 	|
+| Peripherals 	| External components to a computer which can provide input and/or output.  An example is a monitor or keyboard. 	|
+| MAR 	| The Memory Address Register is where the CPU stores an address of an instructor or data in memory.  When data is transferred to or from RAM this is the address used by the CPU.  |
+| MDR | The Memory Data Register is where the CPU stores an information while it is transferred to or from RAM.  |
```

Comment:
Introducing the concept of a page might help clarify this:

```suggestion
| MDR | The Memory Data Register is where the CPU stores a page while it is transferred to or from RAM.  |
```

Reply:
Oh, nevermind this isn't page related.  The MAR and MDR are definitely not things I remember from school, btw.  (We tended to mention registers more generally but stayed at a page level abstraction when talking about copying memory to/from RAM.)